### ISSUE TITLE
Scrub the existing cmdlets and make sure the all offer the -Encrypt, -AccessToken, -TrustServerCertificate, ... parameters

### ISSUE THREAD

**Issue by Matteo-T** [microsoft/SQLServerPSModule#33]
In v22.0.57, I may have missed a few (e.g. Enable-SqlAlwaysOn and Disable-SqlAlwaysOn).

It would be nice to complete the list in some v22 update.

**Comment by potatoqualitee** [comment]
We handle this through a config system. Seems like that'd be a more maintainable solution for you all.

**Comment by Matteo-T** [comment]
What's the definition of "config system"?

**Comment by potatoqualitee** [comment]
In our module, we use Get-DbatoolsConfig and Set-DbatoolsConfig and Register-DbatoolsConfig which saves to a JSON file or the registry. @FriedrichWeinmann created it for us and it's now included in his PSFramework module and is all in C#

We allow users to save the config in their current session (Set) or for all sessions (Register)

```powershell
Set-DbatoolsConfig -Name Import.EncryptionMessageCheck -Value $false -PassThru | Register-DbatoolsConfig
```

Oh, looks like we have way more commands than that, like import and export 😅 

https://dbatools.io/configuration/

HTH

**Comment by AngelaGloner** [comment]
https://github.com/microsoft/opensource.microsoft.com

**Comment by AngelaGloner** [comment]
https://l.facebook.com/l.php?u=https%3A%2F%2Fopensource.microsoft.com%2F%3Ffbclid%3DIwAR2q26arMFIpppiITep6-R_n0UOZ-dVjUsKYiieGw7-Qa7VPtkv_jjLvSrE&h=AT0_VaA3Xt50JMtMiQR38mbcjL-uKOpuhkDHJC2z_c4M-SgoJDfu11cZvDqNs170eriFa1TthGt3dF0xv7uq5McfiVNs4WC3jmctlMb8sASxN907bVg5dnQ1RZLc7UGSz-UaD_bM4tUinJPrMbw5&__tn__=R]-R&c[0]=AT33FnA32YmW1ugYFXRJENf6yyoTANVIaKMXDEyVro9kVQyZe4oeavvv5lLyrxz7F-nrI8Hw4rH5U_EuSDe7_phxA5u_IvWJnnz_1xf-r0G3XRwgDRvXVyrX21khGHJqwSN645pF3IaZGkaF3RXDbV7_

**Comment by potatoqualitee** [comment]
omg where can i report this bot

Edit: reported as malware

**Comment by potatoqualitee** [comment]
This is how we handle it @Matteo-T 

https://blog.netnerds.net/2023/03/new-defaults-for-sql-server-connections-encryption-trust-certificate/

```powershell
# Set defaults just for this session
Set-DbatoolsConfig -FullName sql.connection.trustcert -Value $true
Set-DbatoolsConfig -FullName sql.connection.encrypt -Value $false

# Set these defaults for all future sessions on this machine
Set-DbatoolsConfig -FullName sql.connection.trustcert -Value $true -Register
Set-DbatoolsConfig -FullName sql.connection.encrypt -Value $false -Register
```

**Comment by AngelaGloner** [comment]
https://developer.microsoft.com/en-us/?fbclid=IwAR19nUnFeGRR4FgvJFo_28bNEDJlcIt8fsNjEBmPTyFjdmtt9E4zg2DinMo

**Comment by Matteo-T** [comment]
Yeah - I was thinking about having something like that while talking to the `go-sqcmd` folks. Maybe I'll revisit and add it.

**Comment by potatoqualitee** [comment]
bravo GH and Microsoft, that bot is gone  🙌🏼

**Comment by potatoqualitee** [comment]
feel free to take our design, @Matteo-T. We also include the things found in the SSMS connection panel.

**Comment by Matteo-T** [comment]
> bravo GH and Microsoft, that bot is gone 🙌🏼

Actually, I just deleted those bot-things myself. :-)

**Comment by SQLDBAWithABeard** [comment]
> feel free to take our design, @Matteo-T. We also include the things found in the SSMS connection panel.

I'm pretty sure that this goes for any dbatools/dbachecks code @Matteo-T - No need to reinvent the wheel.

**Comment by Pxtl** [comment]
This would be glorious.  Being able to register "okay, for this database, use these connection settings going forwards" would be wonderful.

I just updated my team's internal devops infrastructure powershell module that we use for all our standard devops operations.  Properly exposing the new `-TrustServerCertificate` param required a change that's -170 +270.  Being able to just say "okay, these are internal development machines and failovers and don't have public certs, they have to be trusted" vs "this is the public prod server, you should verify the cert chain before connecting" in a single config site would be ideal.
